pkg/liqo-controller-manager/storageprovisioner/remoteprovisioner.go

```diff
@@ -97,12 +97,19 @@ func ProvisionRemotePVC(ctx context.Context,
 }
 
 // remotePersistentVolumeClaim forges the apply patch for the reflected PersistentVolumeClaim, given the local one.
-func remotePersistentVolumeClaim(virtualPvc *corev1.PersistentVolumeClaim,
+func remotePersistentVolumeClaim(virtualPvc *corev1.PersistentVolumeClaim, remotePvc *corev1.PersistentVolumeClaim,
 	storageClass, namespace string) *v1apply.PersistentVolumeClaimApplyConfiguration {
-	return v1apply.PersistentVolumeClaim(virtualPvc.Name, namespace).
+	applyConfigurations := v1apply.PersistentVolumeClaim(virtualPvc.Name, namespace).
 		WithLabels(virtualPvc.GetLabels()).
 		WithLabels(forge.ReflectionLabels()).
+		WithAnnotations(filterAnnotations(virtualPvc.GetAnnotations())).
```

Comment:
Objects retrieved from a lister should be deep copied before being mutated.

Reply:
the local PVC is already deep copied in the reflector handler, so this should not be the lister one